adapter/abstract.php

```diff
@@ -50,6 +62,17 @@ public function setPath($path)
 
 		$this->_metadata = null;
 
+        $parts = explode('://', $this->_path);
+
+        if (count($parts) == 2)
+        {
+            if (in_array($parts[0], stream_get_wrappers())) {
```

Comment:
@amazeika this may be silly but can you add a special case for 'file' here with or without stream_get_wrappers. There might be a server out there that returns empty array for stream wrappers

Reply:
@ercanozkaya That's correct inded. Will re-factor this to take this into account.